shared-libs/cht-datasource/src/libs/person.ts

```diff
@@ -0,0 +1,9 @@
+import { v1 as doc } from './doc';
+const { isUuidIdentifier } = doc;
+
+export const validateIdentifier = (identifier: unknown): identifier is V1.UuidIdentifier => {
+  if (isUuidIdentifier(identifier)) {
+    return true;
+  }
+  throw new Error('Invalid identifier');
+};
```

Comment:
```suggestion
// eslint-disable-next-line func-style
export function validateIdentifier(identifier: unknown): asserts identifier is V1.UuidIdentifier {
  if (!isUuidIdentifier(identifier)) {
    throw new Error('Invalid identifier');
  }
}
```

we can leverage typescript's [assertion functions](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#assertion-functions) to nudge the control flow analysis and make it understand more easily the type of the variable
it needs to be defined as a regular `function` though, assertion functions [don't support arrow functions](https://github.com/microsoft/TypeScript/issues/34523)

Reply:
Ahh fancy!  I did not know about assertion functions!

Also looks like things work with arrow functions if you give enough context in the signature:

```
  const assertPersonQualifier: (qualifier: unknown) => asserts qualifier is UuidQualifier = (qualifier: unknown) => {
    if (!isUuidQualifier(qualifier)) {
      throw new Error(`Invalid identifier [${JSON.stringify(qualifier)}].`);
    }
  }
```